admob/src/common/admob_common.cc

```diff
@@ -51,6 +51,41 @@ DEFINE_FIREBASE_VERSION_STRING(FirebaseAdMob);
 static CleanupNotifier* g_cleanup_notifier = nullptr;
 const char kAdMobModuleName[] = "admob";
 
+// Error messages used for completing futures. These match the error codes in
+// the AdMobError enumeration in the C++ API.
+const char* kAdAlreadyInitializedErrorMessage = "Ad is already initialized.";
+const char* kAdCouldNotParseAdRequestErrorMessage =
+    "Could Not Parse AdRequest.";
+const char* kAdLoadInProgressErrorMessage = "Ad is currently loading.";
+const char* kAdUninitializedErrorMessage = "Ad has not been fully initialized.";
+
+// AdListener
+// Default no-op implementation for each listener method so that applications
+// need only implement the methods they're interested in.
+AdListener::~AdListener() {}
+void AdListener::OnAdClicked() {}
```

Comment:
Aside from the destructor (which needs to be here), WDYT about inlining the other implementations so it's clear to developers writing subclasses that they don't need to call the base class methods?

Reply:
Sure!  I'll put this in the next change as well, to reduce merge conflicts.